app/models/service_interaction.rb

```diff
@@ -1,2 +1,4 @@
 class ServiceInteraction < ActiveRecord::Base
+  validates :service_identifier, :service_label, :interaction_identifier,
+  :interaction_label, presence: true, uniqueness: true
```

Comment:
This validation says that `service_identifier`, `service_label`, `interaction_identifier`, and `interaction_label` should all be present and should all be unique.  The first part is correct, but what we really want with uniqueness checking is not that the individual parts are unique, but that the combination is unique.  E.g. we should allow a mapping between one service (pay your council tax) and multiple interactions (find out about, pay, appeal) and we should allow the same interaction (find out about) to be mapped to multiple services (pay your council tax, dog wardens, school places).

Happily, rails gives us a `scope` option on the uniqueness validation to define saw that an attribute should be unique in combination with another, so we'll want to use that.  The [rails guides for the uniqueness validation](http://guides.rubyonrails.org/active_record_validations.html#uniqueness) are a good starting point for learning more.